src/pcre2_compile.c

```diff
@@ -2093,10 +2093,29 @@ else
 
       else
         {
-        c = 0;
-        if (ptr >= ptrend || (cc = XDIGIT(*ptr)) == 0xff) break;  /* Not a hex digit */
+        /* Perl has the surprising/broken behaviour that \x without following
+        hex digits is treated as an escape for NUL. Their source code laments
+        this but keeps it for backwards compatibility. A warning is printed
+        when "use warnings" is enabled, and it's forbidden when "use strict"
+        is enabled. Because we don't have warnings, we simply forbid it. */
```

Comment:
this comment needs updating so it correctly states `use re "strict"`

Reply:
oh dear, didn't realize that the "strict" confusion was also reflected in the documentation.

To clarify, Perl also shows warnings when using less than hexadecimal digits (ex:`/\xa/`), and the warning we are referring to also apply to `/\x/`.

That is documented below, but in the case of the single digit hexadecimals we are ignoring those warnings for backward compatibility.

Again, I am not arguing there should be any changes on behaviour to this code, just pointing out that there is some slight confusion on the rationale for doing them. Let me know if I should instead provide a PR, which might be also a less confusing way to make that argument, it would seem.